tiledb/sm/cpp_api/ndrectangle.h

```diff
@@ -101,13 +101,18 @@ class NDRectangle {
    * ndrect.set_range(0, start, end);
    * @endcode
    *
-   * @tparam T The dimension datatype.
+   * @tparam T The dimension datatype. Must be an integer or a floating point
+   * number.
    * @param dim_name The name of the dimension to add the range to.
    * @param start The range start to add.
    * @param end The range end to add.
    * @return Reference to this NDRectangle.
    */
-  template <class T>
+  template <
+      class T,
+      std::enable_if_t<
```

Comment:
We should not be using `enable_if` in new code. All such binding should be done through `concept`.